src/ome_zarr_models/_rfc5_transforms/multiscales.py

```diff
@@ -0,0 +1,247 @@
+from __future__ import annotations
+
+from typing import Self
+
+from pydantic import (
+    BaseModel,
+    Field,
+    JsonValue,
+    field_validator,
+    model_validator,
+)
+
+from ome_zarr_models._rfc5_transforms.coordinate_transformations import (
+    CoordinateSystem,
+    CoordinateTransformationType,
+    Scale,
+    Translation,
+)
+from ome_zarr_models.base import BaseAttrs
+from ome_zarr_models.common.validation import check_length
+
+__all__ = ["Dataset"]
+
+
+class Dataset(BaseAttrs):
+    """
+    An element of Multiscale.datasets.
+    """
+
+    path: str
+    # TODO: suggest John not to have a tuple for the coordinate transformations of a
+    #  dataset, since it's either a single scale or a single sequence (with a scale
+    #  and a translation)
+    coordinateTransformations: tuple[CoordinateTransformationType, ...] = Field(
+        ..., min_length=1, max_length=1
+    )
+
+    # the before validation is used to simplify the error messages
+    @field_validator("coordinateTransformations", mode="before")
+    def _ensure_scale_translation(
+        transforms_obj: object,
+    ) -> object:
+        """
+        Ensures that
+        - a single transformation is present
+        - such transformation is a scale
+        - if such transformation is a sequence, ensure that its length is 2 and that
+          the first transformation is a scale and the second a translation
+        """
+
+        class Transforms(BaseModel):
+            transforms: tuple[CoordinateTransformationType, ...]
+
+        transforms = Transforms(transforms=transforms_obj).transforms
```

Comment:
this is too many "transforms" 😆 are you just using this class for validation? because that's why it's better IMO to define validator functions separately from the classes, so you can re-use them.

Reply:
I agree, quite a weird line and having a helper function would be better. The problem is that the two `class Transform` are using different `CoordinateTransformation` classes (one from `common` and the other one from `_rfc5_transforms`). So in practice, maybe the current is already one of the simplest/shortest codes we can have.

I'd leave it as it is, but if you want to try a simplification please you are welcome 😊